common/src/test/java/io/pravega/common/util/SimpleCacheTests.java

```diff
@@ -495,4 +499,12 @@ private long value(long key) {
         Collections.shuffle(list, random);
         return list;
     }
+    
+    public static class TestSimpleCache<KeyT, ValueT> extends SimpleCache<KeyT, ValueT> {
```

Comment:
What is the point of this?

Reply:
I see where you use this. Please consider using `moq` (i.e., `mock(SimpleCache.class)`) if you need to mock its behavior - that will also make your unit test more focused and not rely on the `SimpleCache` behavior.

Reply:
`SimpleCache` does not provide a mechanism to set the supplier which is used as a clock. In unit tests, the clock has to be manipulated for setting time.  To do that the package private constructor has to be used. Hence a new class has been created for testing with a public constructor to which the supplier for clock can be provided.

Reply:
I realized that as I went further down this PR. I suggested using some mocks to help with your unit tests. Have you tried that?  I prefer to leave adding new dependencies as a last resort as they are very hard to undo and may complicate refactoring work later on.

Reply:
Have mocked the cache.  Added dependency has been removed.